components/data-table/private/header-cell.jsx

```diff
@@ -172,7 +182,17 @@ const DataTableHeaderCell = createReactClass({
 						) : null}
 					</a>
 				) : (
-					<div className="slds-truncate">{label}</div>
+					<div
+						className={classNames(
+							{
+								// Adds padding to header cells if the are not sortable, but are in a fixed layour / advanced table.
+								'slds-p-horizontal_x-small': this.props.fixedLayout,
```

Comment:
It looks like the SLDS sample has its padding coming from `<a class="slds-th__action">`.  I think this is ok for us though since ours isn't clickable, and I couldn't find a sample for this case.  All their advanced tables have clickable header cells.